x-pack/test_serverless/shared/services/user_manager/saml_auth.ts

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import axios, { AxiosResponse } from 'axios';
+import Url from 'url';
+import * as cheerio from 'cheerio';
+import { parse as parseCookie } from 'tough-cookie';
+import { createSAMLResponse as createMockedSAMLResponse } from '@kbn/mock-idp-plugin/common';
+import { ToolingLog } from '@kbn/tooling-log';
+import { Session } from './svl_user_manager';
+
+export interface SAMLSessionParams {
+  username: string;
+  password: string;
+  kbnHost: string;
+  kbnVersion: string;
+  log: ToolingLog;
+}
+
+export interface FakeSAMLSessionParams {
+  username: string;
+  email: string;
+  fullname: string;
+  role: string;
+  kbnHost: string;
+  log: ToolingLog;
+}
+
+const getSessionCookie = (cookieString: string) => {
+  return parseCookie(cookieString);
+};
+
+const getCloudHostName = () => {
+  const hostname = process.env.TEST_CLOUD_HOST_NAME;
+  if (!hostname) {
+    throw new Error('SAML Authentication requires TEST_CLOUD_HOST_NAME env variable to be set');
+  }
+
+  return hostname;
+};
+
+const getCloudUrl = (hostname: string, pathname: string) => {
+  return Url.format({
+    protocol: 'https',
+    hostname,
+    pathname,
+  });
+};
+
+const createCloudSession = async (hostname: string, email: string, password: string) => {
+  const cloudLoginUrl = getCloudUrl(hostname, '/api/v1/users/_login');
+  const sessionResponse: AxiosResponse = await axios.request({
+    url: cloudLoginUrl,
+    method: 'post',
+    data: {
+      email,
+      password,
+    },
+    headers: {
+      accept: 'application/json',
+      'content-type': 'application/json',
+    },
+    validateStatus: () => true,
+    maxRedirects: 0,
+  });
+  const firstName = sessionResponse?.data?.user?.data?.first_name ?? '';
+  const lastName = sessionResponse?.data?.user?.data?.last_name ?? '';
+  const fullname = `${firstName} ${lastName}`.trim();
+  return {
+    token: sessionResponse?.data?.token as string,
```

Comment:
__optional nit:__ just to have a better debugging experience should Cloud API change:
```ts
  const token = sessionResponse?.data?.token as string;
  if (!token) {
    // And log maybe? Not sure what is captured in CI logs.
    throw new Error(`Unable to create Cloud session, token is missing.`);
  }
```

Reply:
Good point, added in 2b93cec1296df6068c577d2222a5718756a8b94b